graphdatascience/graph/graph_entity_ops_runner.py

```diff
@@ -85,6 +89,25 @@ def stream(
                 columns={"variable": "nodeProperty", "value": "propertyValue"}
             )
 
+        if db_node_properties:
+            nodeIds = result["nodeId"].tolist()
+            db_nodes = self._util_proc_runner.asNodes(nodeIds)
+            properties = []
+            for node in db_nodes:
+                properties.append(
+                    list([node.id] + list(map(lambda p: node.get(p), db_node_properties)))  # type: ignore
+                )
+            db_properties_df = pd.DataFrame(properties, columns=["nodeId"] + db_node_properties).drop_duplicates(
+                subset=["nodeId"]
+            )
+            if "propertyValue" not in result.keys():
+                result = result.join(db_properties_df.set_index("nodeId"), on="nodeId")
+            else:
+                db_properties_df = db_properties_df.melt(id_vars=["nodeId"]).rename(
```

Comment:
we only need melt as you return the whole node objects. So just returning the desired properties + nodeId would make this part also easier :)